### PR TITLE
Rst 6242 replace print statements with logger objects in admin app & grc app

### PR DESCRIPTION
**Pull request by sgmselli**

### CODE REVIEW

File: admin/jobs/anonymise_expired_applications.py [hunk 1]
```diff
@@ -4,13 +4,15 @@
 from flask.cli import with_appcontext
 from grc.models import db, Application, ApplicationStatus
 from grc.utils.application_files import ApplicationFiles
+from grc.utils.logger import LogLevel, Logger
 
+logger = Logger()
```
Comment (itoulou) [inline admin/jobs/anonymise_expired_applications.py:9]: I wouldn't use the logger in the jobs. The logger uses the flask session data and jobs won't be in a session so it will complain about working outside the app context. I'd revert these back to print statements in ALL the jobs.

File: admin/jobs/delete_expired_security_codes.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 
 def delete_security_codes():
-    print(f'\nDeleting expired security codes', flush=True)
```
Comment (itoulou) [inline admin/jobs/delete_expired_security_codes.py:11]: The flush arg is important to make sure the buffered output is logged out to the console at the time of printing.

File: grc/models.py [hunk 3]
```diff
@@ -6,9 +6,11 @@
 from sqlalchemy_utils import StringEncryptedType
 from sqlalchemy_utils.types.encrypted.encrypted_type import AesEngine
 from grc.business_logic.data_structures.application_data import ApplicationData
+from grc.utils.logger import LogLevel, Logger
 
 db = SQLAlchemy()
 secret_key = os.environ.get('SQLALCHEMY_KEY', '')
+logger = Logger()
```
Comment (itoulou) [inline grc/models.py:13]: I wouldn't add to the models either for similar reasons as my comment regarding the jobs

File: grc/utils/flask_child_form_add_custom_errors.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
 from flask_wtf import FlaskForm
 from wtforms import FieldList
+from grc.utils.logger import LogLevel, Logger
 
+logger = Logger()
 
 def add_error_for_child_form(parent_form_field_list: FieldList, child_form: FlaskForm, child_form_field_name: str, error_message: str):
-    print(f"field_list.errors: {parent_form_field_list.errors}", flush=True)
+    logger.log(LogLevel.WARN, f"field_list.errors: {parent_form_field_list.errors}")
```
Comment (itoulou) [inline grc/utils/flask_child_form_add_custom_errors.py:8]: I think these can just be removed completely.

File: tests/jobs/conftest.py [hunk 5]
```diff
@@ -32,7 +32,8 @@ def test_started_and_expired_applications(app, public_user_email):
             new_app.status = ApplicationStatus.STARTED
             new_app.updated = datetime.now() - relativedelta(days=200)
             db.session.commit()
-            print(f'Test Inactive App Ref - {new_app.reference_number}', flush=True)
```
Comment (itoulou) [inline tests/jobs/conftest.py:35]: The flush arg is important to make sure the buffered output is logged out to the console at the time of printing.

File: tests/jobs/test_delete_completed_applications.py [hunk 6]
```diff
@@ -1,7 +1,10 @@
 import pytest
 from grc.models import db, Application, ApplicationStatus, ApplicationData
+from grc.utils.logger import LogLevel, Logger
 from unittest.mock import patch, MagicMock
 
+logger = Logger()
```
Comment (itoulou) [inline tests/jobs/test_delete_completed_applications.py:6]: This is not being used

Comment (itoulou) [reply tests/jobs/test_delete_completed_applications.py:6]: I assume you wanted to use it for line 30. Think that print statement can just be removed from the test

File: grc/utils/flask_child_form_add_custom_errors.py (hunk 4 above)
Comment (sgmselli) [reply grc/utils/flask_child_form_add_custom_errors.py:8]: The print statements as well?

File: admin/jobs/delete_completed_applications.py
Comment (itoulou) [inline admin/jobs/delete_completed_applications.py:1]: Still logger objects in this job which need to be removed

File: admin/jobs/delete_expired_security_codes.py
Comment (itoulou) [inline admin/jobs/delete_expired_security_codes.py:1]: Still logger objects in this job which need to be removed

File: admin/jobs/delete_expired_security_codes.py [hunk 7]
```diff
@@ -43,6 +43,5 @@ def main():
     except Exception as e:
         print(f'Error delete expired security codes cron, message = {e}', flush=True)
 
-
```
Comment (itoulou) [inline admin/jobs/delete_expired_security_codes.py:46]: This change isn't needed

File: admin/jobs/notify_applicants_inactive_apps.py
Comment (itoulou) [inline admin/jobs/notify_applicants_inactive_apps.py:1]: Still logger objects in this job which need to be removed

File: setup-db.py
Comment (itoulou) [inline setup-db.py:1]: This file change is out of scope for this ticket

File: tests/jobs/test_delete_completed_applications.py
Comment (itoulou) [inline tests/jobs/test_delete_completed_applications.py:1]: This logger object still not being used and can be removed

File: tests/test_gov_uk_design_system_folders_in_sync.py [hunk 8]
```diff
@@ -2,7 +2,6 @@
 import pathlib
 import pytest
 
-
```
Comment (itoulou) [inline tests/test_gov_uk_design_system_folders_in_sync.py:5]: Change not needed

File: tests/test_ping_json.py [hunk 9]
```diff
@@ -24,6 +24,7 @@ def test_json_health_check(max_retries, timeout):
 
         except requests.RequestException as e:
             print(f"Attempt {attempt} failed: {e}")
+
```
Comment (itoulou) [inline tests/test_ping_json.py:27]: Change not needed

**Review by itoulou — APPROVED** [review]
